evm/src/executor/inspector/cheatcodes/ext.rs

```diff
@@ -296,6 +297,33 @@ fn remove_file(state: &mut Cheatcodes, path: impl AsRef<Path>) -> Result<Bytes,
     Ok(Bytes::new())
 }
 
+/// Gets the metadata of a file/directory
+///
+/// This will return an error if no file/directory is found, or if the target path isn't allowed
+fn fs_metadata(state: &mut Cheatcodes, path: impl AsRef<Path>) -> Result<Bytes, Bytes> {
+    let path =
+        state.config.ensure_path_allowed(&path, FsAccessKind::Read).map_err(error::encode_error)?;
+
+    let metadata = path.metadata().map_err(error::encode_error)?;
+
+    // These fields not available on all platforms; default to 0
+    let [modified, accessed, created] =
+        [metadata.modified(), metadata.accessed(), metadata.created()].map(|time| {
+            time.unwrap_or(UNIX_EPOCH).duration_since(UNIX_EPOCH).unwrap_or_default().as_secs()
+        });
```

Comment:
Defaulting to zero makes sense to me since it follows solidity's conventions. Only comment would be to document which platforms don't support these fields (no changes required in this PR, we can document that in the book)